src/modules/roc_core/optional.h

```diff
@@ -85,6 +85,11 @@ template <class T> class Optional : public NonCopyable<> {
         MaxAlign align;
         char mem[sizeof(T)];
     };
+    void destroy_() {
```

Comment:
Nitpick: missing empty line.

Also, since destroy_() is short and used only once, I'd inline it into reset().

Reply:
Yeah you're right it is better now 👍